internal/schedule/schedule.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2025 Stefan Prodan.
+// SPDX-License-Identifier: AGPL-3.0
+
+package schedule
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/robfig/cron/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	fluxcdv1 "github.com/controlplaneio-fluxcd/flux-operator/api/v1"
+)
+
+const (
+	// cronLookBack is the maximum time range a previous cron trigger
+	// could have occurred in the past. It's 5 years because a cron
+	// schedule can be defined for Feb 29, which only occurs every 4
+	// years (0 0 29 2 *).
+	cronLookBack = 5 * 365 * 24 * time.Hour
+)
+
+// Schedule computes when reconciliations should happen according
+// to a list of schedule configurations.
+type Schedule struct {
+	schedules []schedule
+	timeout   time.Duration
+}
+
+type schedule struct {
```

Comment:
This name collides with the package name, please rename.

Reply:
Fixed, thanks!